modules/infinite_blocks/src/Plugin/Block/InfiniteHeaderMediaBlock.php

```diff
@@ -20,9 +20,10 @@ class InfiniteHeaderMediaBlock extends BlockBase {
    * {@inheritdoc}
    */
   public function build() {
+    $entity = NULL;
     if ($node = \Drupal::request()->attributes->get('node')) {
       /* @var Node $node */
-      $entity = $node;
+      $entity = is_object($node) ? $node : Node::load($node);
```

Comment:
Yes. I'm going to try to load the requested revision. I close the PR and re-open it when it's implemented.